lib/transport/tls-context.c

```diff
@@ -460,6 +460,40 @@ tls_context_load_key_and_cert(TLSContext *self)
   return TLS_CONTEXT_OK;
 }
 
+static gboolean
+_set_client_ca_list(TLSContext *self)
+{
+#if SYSLOG_NG_HAVE_DECL_SSL_ADD_DIR_CERT_SUBJECTS_TO_STACK && SYSLOG_NG_HAVE_DECL_SSL_ADD_FILE_CERT_SUBJECTS_TO_STACK
+
+  STACK_OF(X509_NAME) *ca_list = sk_X509_NAME_new_null();
+  const STACK_OF(X509_NAME) *existing_ca_list = SSL_CTX_get_client_CA_list(self->ssl_ctx);
+
+  for (gint i = 0; i < sk_X509_NAME_num(existing_ca_list); i++)
+    {
+      sk_X509_NAME_push(ca_list, X509_NAME_dup(sk_X509_NAME_value(existing_ca_list, i)));
+    }
+
+  if (_is_file_accessible(self, self->ca_dir) && !SSL_add_dir_cert_subjects_to_stack(ca_list, self->ca_dir))
+    {
+      sk_X509_NAME_free(ca_list);
+      return FALSE;
+    }
+
+  if (_is_file_accessible(self, self->ca_file) && !SSL_add_file_cert_subjects_to_stack(ca_list, self->ca_file))
+    {
+      sk_X509_NAME_free(ca_list);
+      return FALSE;
+    }
```

Comment:
Can we add the CAs that we parser from a PKCS #12 archive when `pkcs12-file()` is used?

Reply:
They are already added: https://github.com/syslog-ng/syslog-ng/pull/4412/files#diff-8f4f2584202391a2f25b8211042a90744d54bbcedc03da6a60eeadbe1f2cd4d3R383

I believe they are kept as I am getting, copying, appending, then setting the CA list. However, I did not have the time to check it myself with pkcs12. I will do that tomorrow.

Reply:
Oh, I did not remember that. Thanks.

Reply:
Tested the `pkcs12-file()` with `ca-dir()`, and they work together well.
![image](https://user-images.githubusercontent.com/45236571/229715438-a566e59d-fe73-444b-aac3-f3d9981c05f2.png)

Reply:
Updated the news entry :)

Reply:
Cool. Thank you.